### PR TITLE
feat: impl sequence batches

### PR DESCRIPTION
**Pull request by captainlee1024**

### CODE REVIEW

File: src/batch_proposer/mod.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ use std::time::Duration;
 use tokio::sync::broadcast;
 use tokio::{select, time};
 
-const L2_WATCHER_INTERVAL: Duration = Duration::from_secs(1);
+const L2_WATCHER_INTERVAL: Duration = Duration::from_secs(30);
```
Comment (eigmax) [inline src/batch_proposer/mod.rs:10]: 30 is too long, can we set it to 2?

File: src/settlement/worker.rs [hunk 2]
```diff
@@ -205,4 +212,176 @@ impl Settler {
...
+    }
+}
+
+pub fn encode_fields(tx: &TxLegacy, out: &mut dyn bytes::BufMut) {
+    tx.nonce.encode(out);
+    tx.gas_price.encode(out);
+    tx.gas_limit.encode(out);
+    tx.to.encode(out);
+    tx.value.encode(out);
+    tx.input.0.encode(out);
```
Comment (eigmax) [inline src/settlement/worker.rs:401]: Why do we encode input 0 only?

File: src/settlement/worker.rs [hunk 3]
```diff
@@ -205,4 +212,176 @@ impl Settler {
             }
         }
     }
+
+    pub(crate) async fn submit_worker(
```
Comment (eigmax) [inline src/settlement/worker.rs]: We can change it to rollup

File: src/settlement/worker.rs (hunk 2 above)
Comment (captainlee1024) [reply src/settlement/worker.rs:401]: input is a tuple struct that wraps a Bytes type.  `pub struct Bytes(pub bytes::Bytes);`
input.0 contains all data of the input field.

File: src/batch_proposer/mod.rs (hunk 1 above)
Comment (captainlee1024) [reply src/batch_proposer/mod.rs:10]: ok

File: src/settlement/worker.rs (hunk 3 above)
Comment (captainlee1024) [reply src/settlement/worker.rs]: ok

**Pull request merged**